docs/src/interface.md

```diff
@@ -64,3 +57,49 @@ julia> y[1]
 
 * [`allocate_result`](@ref) allocates a result of a particular function (for example [`exp`], [`flat`], etc.) on a particular manifold with particular arguments.
   It takes into account the possibility that different arguments may have different numeric [`number_eltype`](@ref) types thorough the [`ManifoldsBase.allocate_result_type`](@ref) function.
+
+## A Decorator for manifolds
+
+ decorator manifold extends the functionality of a [`Manifold`](@ref) in a semi-transparent way.
+It internally stores the [`Manifold`](@ref) it extends and by default for functions defined in the [`ManifoldsBase`](interface.md) it acts transparently in the sense that it passes all functions through to the base except those that it actually affects.
+For example, because the [`ArrayManifold`](@ref) affects nearly all functions, it overwrites nearly all functions, except a few like [`manifold_dimension`](@ref).
+On the other hand, the [`MetricManifold`](@ref) only affects functions that involve metrics, especially [`exp`](@ref) and [`log`](@ref) but not the [`manifold_dimension`](@ref).
```

Comment:
The main difference here is that `ArrayManifold` overwrites fallback definitions while `MetricManifold` does not.

Reply:
I made that more specific.